### PR TITLE
Implementing disqus widget to show comments on blog posts

### PR DESCRIPTION
**Pull request by monkemedia**

### CODE REVIEW

File: pages/blog/_.vue [hunk 1]
```diff
@@ -24,6 +24,14 @@
           v-if="page.keywords"
           :tags="page.keywords"
         />
+
+        <div class="comments mt-4">
```
[the remark this answers is not in the document]
Comment (monkemedia) [reply pages/blog/_.vue]: Done

File: pages/blog/_.vue [hunk 2]
```diff
@@ -72,6 +80,16 @@
       },
       heroImage() {
         return this.hero ? this.hero.image.fields.file : null;
+      },
+
+      pageUrl() {
+        const host = 'https://www.europeana.eu';
```
[the remark this answers is not in the document]
Comment (monkemedia) [reply pages/blog/_.vue]: Done

File: pages/blog/_.vue [hunk 3]
```diff
@@ -72,6 +84,14 @@
...
         return this.hero ? this.hero.image.fields.file : null;
+      },
+
+      ...mapGetters({
+        shareUrl: 'http/canonicalUrl'
+      }),
+
+      identifier() {
+        return this.$route.params.pathMatch;
       }
```
Comment (lbiedinger) [inline pages/blog/_.vue:95]: This identifer will need to include the hostname and something to identify the fact it's going on a blog:
```   return this.origin + '/blog/' + this.$route.params.pathMatch;```
You'll have to include the 'origin' getter from http as you do with thte canonicalUrl.

File: .env.example [hunk 4]
```diff
@@ -71,3 +71,8 @@ ENABLE_LINKS_TO_CLASSIC=0
 
 # Retrieve and display tagging annotations on item pages.
 ENABLE_ITEM_TAGGING_ANNOTATIONS=0
+
+# If enabled, shows comments below blog posts.
+ENABLE_BLOG_COMMENTS=1
+# Name given to you by Disqus. https://help.disqus.com/en/articles/1717111-what-s-a-shortname
+DISQUS_SHORTNAME=www-europeana-eu
```
Comment (lbiedinger) [inline .env.example:78]: Add a dummy value here.

File: pages/blog/_.vue [hunk 5]
```diff
@@ -24,6 +24,17 @@
           v-if="page.keywords"
           :tags="page.keywords"
         />
+
+        <div
+          class="card card-body mt-4"
+          data-qa="disqus widget"
+        >
+          <vue-disqus
+            shortname="www-europeana-eu"
```
[the remark this answers is not in the document]
Comment (lbiedinger) [reply pages/blog/_.vue:34]: This value needs to come from the DISQUS_SHORTNAME variable/setting

File: pages/blog/_.vue [hunk 6]
```diff
@@ -24,6 +24,18 @@
           v-if="page.keywords"
           :tags="page.keywords"
         />
+
+        <div
+          v-if="enableBlogComments"
```
Comment (lbiedinger) [inline pages/blog/_.vue:29]: Get rid of the 'ENABLE_BLOG_COMMENTS' variable from the .env file
Instead have this be conditional on DISQUS_SHORTNAME not being undefined, that can be loaded into a variable where ` enableBlogComments()` is now. Then just check that variable here.

File: pages/blog/_.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (monkemedia) [reply pages/blog/_.vue:34]: Whoops

File: pages/blog/_.vue [hunk 7]
```diff
@@ -72,6 +85,23 @@
...
         return this.hero ? this.hero.image.fields.file : null;
+      },
+
+      ...mapGetters({
+        shareUrl: 'http/canonicalUrl',
+        origin: 'http/origin'
+      }),
+
+      identifier() {
+        return `${this.origin}/blog/${this.$route.params.pathMatch}`;
```
Comment (rwd) [inline pages/blog/_.vue:96]: This should really be delegated to a reusable Vuex getter in the http store, e.g.
```js
  canonicalUrlWithoutLocale(state, getters) {
    return getters.canonicalUrl.replace(/(:\/\/[^/]+)\/[a-z]{2}(\/|$)/, '$1$2');
  }
```

**Pull request merged**